### PR TITLE
BUG: loffset has no effect when passing in a numpy.timedelta64

### LINKED ISSUE #7687: loffset has no effect when passing in a numyp.timedelta64
From mailing-list: [https://groups.google.com/forum/#!topic/pydata/CDnF9cNR2ho]
When doing a re-sample, loffset has no effect if a numpy.timedelta64 is passed in:

``` python
In [1]: rng = pd.date_range('1/1/2012', periods=4, freq='S')

In [2]: ts = pd.Series(np.random.randint(0, 500, len(rng)), index=rng)

In [3]: ts.resample('2S')
Out[3]:
2012-01-01 00:00:00    396.0
2012-01-01 00:00:02    214.5
Freq: 2S, dtype: float64

In [4]: import datetime

In [5]: ts.resample('2S',loffset=datetime.timedelta(seconds=10))
Out[5]:
2012-01-01 00:00:10    396.0
2012-01-01 00:00:12    214.5
dtype: float64

In [6]: ts.resample('2S',loffset=np.timedelta64(10,'s'))
Out[6]:
2012-01-01 00:00:00    396.0
2012-01-01 00:00:02    214.5
Freq: 2S, dtype: float64
```

### PR DESCRIPTION
**Pull request by discort**
- [x] closes #7687
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/resample.py [hunk 1]
```diff
@@ -365,8 +365,9 @@ def _apply_loffset(self, result):
             the result of resample
         """
 
+        loffset_types = (DateOffset, timedelta, np.timedelta64)
```
Comment (discort) [inline pandas/core/resample.py]: `np.timedelta64` is not the instance of `timedelta`

**Comment by discort** [comment]
@TomAugspurger

File: pandas/tests/test_resample.py [hunk 2]
```diff
@@ -1173,27 +1173,19 @@ def test_resample_frame_basic(self):
         df.resample('M', kind='period').mean()
         df.resample('W-WED', kind='period').mean()
 
-    def test_resample_loffset(self):
+    @pytest.mark.parametrize('loffset', [timedelta(minutes=1),
+                                         '1min', Minute(1),
+                                         np.timedelta64(1, 'm')])
+    def test_resample_loffset(self, loffset):
```
Comment (gfyoung) [inline pandas/tests/test_resample.py]: Can we put a comment referencing the issue next to the relevant input value to `loffset`?

**Comment by discort** [comment]
@jreback

**Comment by discort** [comment]
@jreback

**Comment by discort** [comment]
@jreback

File: doc/source/whatsnew/v0.24.0.txt [hunk 3]
```diff
@@ -703,6 +703,7 @@ Groupby/Resample/Rolling
 - Multiple bugs in :func:`pandas.core.Rolling.min` with ``closed='left'` and a
   datetime-like index leading to incorrect results and also segfault. (:issue:`21704`)
 - Bug in :meth:`Resampler.apply` when passing postiional arguments to applied func (:issue:`14615`).
+- Bug in :meth:`Series.resample` when passing numpy.timedelta64 to `loffset` kwarg (:issue:`7687`).
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: double backticks around numpy.timedelta64

File: pandas/core/resample.py [hunk 4]
```diff
@@ -365,8 +365,9 @@ def _apply_loffset(self, result):
...
         needs_offset = (
-            isinstance(self.loffset, (DateOffset, timedelta)) and
```
Comment (jreback) [inline pandas/core/resample.py]: go ahead and inlinethese (add an overall comment if you would)

cc @jbrockmendel we should define scalar function helpers for this as we repeat lots of this in ops....

File: pandas/tests/test_resample.py [hunk 5]
```diff
@@ -1173,27 +1173,20 @@ def test_resample_frame_basic(self):
         df.resample('M', kind='period').mean()
         df.resample('W-WED', kind='period').mean()
 
-    def test_resample_loffset(self):
+    @pytest.mark.parametrize('loffset', [timedelta(minutes=1),
```
Comment (jreback) [inline pandas/tests/test_resample.py]: if you want to use hypothesis would be great here :> (can be followup, not sure how easy)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_resample.py (hunk 5 above)
Comment (discort) [reply pandas/tests/test_resample.py]: @jreback 

There are no many examples in pandas codebase how to use hypothesis. Furthermore, I know only 4 cases which need to test: `timedelta(minutes=1), '1min', Minute(1), np.timedelta64(1, 'm')`.  Did I miss something?

Comment (discort) [reply pandas/tests/test_resample.py]: ping @jreback

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @discort

**Pull request merged**